src/Compilers/VisualBasic/Portable/Syntax/SyntaxNodeFactories.vb

```diff
@@ -33,6 +34,34 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return VisualBasicSyntaxTree.Create(DirectCast(root, VisualBasicSyntaxNode), DirectCast(options, VisualBasicParseOptions), path, encoding)
         End Function
 
+        ''' <summary>
+        ''' Produces a syntax tree by parsing the source text.
+        ''' </summary>
+        Public Shared Function ParseSyntaxTree(
+            text As String,
+            options As ParseOptions,
+            path As String,
+            encoding As Encoding,
+            cancellationToken As CancellationToken) As SyntaxTree
+
+            Return ParseSyntaxTree(SourceText.From(text, encoding), options, path, cancellationToken)
+        End Function
+
+        ''' <summary>
+        ''' Produces a syntax tree by parsing the source text.
+        ''' </summary>
+        Public Shared Function ParseSyntaxTree(
+            text As SourceText,
+            options As ParseOptions,
+            path As String,
+            cancellationToken As CancellationToken) As SyntaxTree
+
+            Return VisualBasicSyntaxTree.ParseText(text, DirectCast(options, VisualBasicParseOptions), path, cancellationToken)
+        End Function
+
+#Disable Warning RS0026 ' Do not add multiple public overloads with optional parameters.
+#Disable Warning RS0027 ' Public API with optional parameter(s) should have the most parameters amongst its public overloads.
+
```

Comment:
Does the analyzer need fixing?

Reply:
Dunno, this was something I was following from @agocke. Andy did something seem up when you did it? It seemed like to me it was less of a problem because there's multiple overloads that both start a with a different type (string vs. SourceText) so there weren't ambiguity problems.